modules/fluid_properties/src/fluidproperties/TemperaturePressureFunctionFluidProperties.C

```diff
@@ -23,8 +23,17 @@ TemperaturePressureFunctionFluidProperties::validParams()
   params.addRequiredParam<FunctionName>(
       "mu", "Dynamic viscosity function of temperature and pressure [Pa-s]");
 
-  params.addRequiredRangeCheckedParam<Real>(
-      "cv", "cv > 0", "Constant isochoric specific heat [J/(kg-K)]");
+  params.addParam<FunctionName>(
+      "cp", "Isobaric specific heat function of temperature and pressure [J/(kg-K)]");
+  params.addRangeCheckedParam<Real>(
+      "cv", 0, "cv >= 0", "Constant isochoric specific heat [J/(kg-K)]");
+  params.addParam<Real>("e_ref", 0, "Specific internal energy at the reference temperature");
+  params.addParam<Real>("T_ref", 0, "Reference temperature for the specific internal energy");
+  params.addParam<unsigned int>(
+      "n_integration_intervals",
```

Comment:
I think `dT_integration_intervals` would be a better parameter so that it's the same "resolution" for all temperatures.